### PR TITLE
feat: Warehouse redesign part2

### PR DESCRIPTION
**Pull request by sfc-gh-asawicki**
Warehouses redesign part2 (continuation of https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/2864):
- state upgraders for all cases (with acceptance tests)
- TODOs in acceptance and integration tests solved
- suspending the resource on the SDK level
- change resource monitor type in the SDK
- add missing UNSET tests
- handle not working UNSETs with proper SETs
- migration guide updated

Other changes:
- Use `Name()` for account object identifiers in show output

Still missing (next PR):
- datasource

Note:
- parts of the logic may be still changed after our internal discussions; it may affect migration guide and the resource behavior

### CODE REVIEW

File: pkg/sdk/warehouses.go [hunk 1]
```diff
@@ -305,10 +305,32 @@ func (c *warehouses) Alter(ctx context.Context, id AccountObjectIdentifier, opts
...
+		if err != nil {
+			return err
+		}
+		if warehouse.State == WarehouseStateStarted {
+			err := c.Alter(ctx, id, &AlterWarehouseOptions{Suspend: Bool(true)})
+			if err != nil {
+				return err
+			}
+			defer func() {
+				_ = c.Alter(ctx, id, &AlterWarehouseOptions{Resume: Bool(true)})
```
Comment (sfc-gh-jcieslak) [inline pkg/sdk/warehouses.go:321]: Maybe we should at least log an error when it occurred after a failed alter

File: pkg/sdk/warehouses.go [hunk 2]
```diff
@@ -483,6 +504,9 @@ func (row warehouseDBRow) convert() *Warehouse {
 	if row.OwnerRoleType.Valid {
 		wh.OwnerRoleType = row.OwnerRoleType.String
 	}
+	if row.ResourceMonitor != "null" {
```
Comment (sfc-gh-jcieslak) [inline pkg/sdk/warehouses.go:507]: `|| row.ResourceMonitor != ""` ?

File: pkg/resources/warehouse_acceptance_test.go [hunk 3]
```diff
@@ -39,124 +44,168 @@ func TestAcc_Warehouse(t *testing.T) {
...
+					resource.TestCheckResourceAttr("snowflake_warehouse.w", "parameters.0.statement_queued_timeout_in_seconds.0.value", "0"),
+					resource.TestCheckResourceAttr("snowflake_warehouse.w", "parameters.0.statement_timeout_in_seconds.0.value", "172800"),
+				),
+			},
+			// IMPORT after empty config (in this method, most of the attributes will be filled with the defaults acquired from Snowflake)
+			{
+				ResourceName: "snowflake_warehouse.w",
+				ImportState:  true,
+				ImportStateCheck: importchecks.ComposeImportStateCheck(
+					importchecks.TestCheckResourceAttrInstanceState(warehouseId.Name(), "warehouse_type", string(sdk.WarehouseTypeStandard)),
```
Comment (sfc-gh-jcieslak) [inline pkg/resources/warehouse_acceptance_test.go:91]: what about the "name" field

File: pkg/sdk/warehouses.go (hunk 2 above)
Comment (sfc-gh-asawicki) [reply pkg/sdk/warehouses.go:507]: Had it before but it is not necessary -> will result in the same struct being created

File: pkg/sdk/warehouses.go (hunk 1 above)
Comment (sfc-gh-asawicki) [reply pkg/sdk/warehouses.go:321]: Oky

Comment (sfc-gh-asawicki) [reply pkg/sdk/warehouses.go:321]: Done.

File: pkg/resources/warehouse_acceptance_test.go (hunk 3 above)
Comment (sfc-gh-asawicki) [reply pkg/resources/warehouse_acceptance_test.go:91]: 🤔 - added

File: pkg/resources/warehouse.go [hunk 4]
```diff
@@ -29,80 +29,86 @@ var warehouseSchema = map[string]*schema.Schema{
...
-		Type:         schema.TypeString,
-		Description:  "Specifies whether to enable the query acceleration service for queries that rely on this warehouse for compute resources.",
-		ValidateFunc: validation.StringInSlice([]string{"true", "false"}, true),
-		Optional:     true,
-		Default:      "unknown",
+		Type:             schema.TypeString,
+		Optional:         true,
+		ValidateFunc:     validation.StringInSlice([]string{"true", "false"}, true),
+		DiffSuppressFunc: IgnoreChangeToCurrentSnowflakeValue("enable_query_acceleration"),
+		Description:      "Specifies whether to enable the query acceleration service for queries that rely on this warehouse for compute resources.",
```
[the remark this answers is not in the document]
Comment (sfc-gh-asawicki) [reply pkg/resources/warehouse.go:102]: You have already asked for this and there is a TODOs if we pick this approach:
```
// TODO [SNOW-1348102 - next PR]: extract three-value logic; add better description for each field
```

File: pkg/resources/warehouse.go [hunk 5]
```diff
@@ -352,6 +358,71 @@ func GetReadWarehouseFunc(withExternalChangesMarking bool) schema.ReadContextFun
 			}
 		}
 
+		// These are all identity sets, needed for the case where:
+		// - previous config was empty (therefore Snowflake defaults had been used)
+		// - new config have the same values that are already in SF
+		if !d.GetRawConfig().IsNull() {
+			if v := d.GetRawConfig().AsValueMap()["warehouse_type"]; !v.IsNull() {
```
[the remark this answers is not in the document]
Comment (sfc-gh-asawicki) [reply pkg/resources/warehouse.go:365]: If we pick this approach we can make it prettier, I don't think it's the right moment to do so

**Review by sfc-gh-jcieslak — APPROVED** [review]

**Pull request merged**